### DISCUSSION TITLE
Tanstack Table: Best practices for handling table accessor patterns

### DISCUSSION THREAD

**Discussion by ekegus** [0no-co/gql.tada#419] [Q&A]
I'm encountering "Field X is not used" warnings when using gql-tada with @tanstack/react-table. The fields are being accessed dynamically through table column accessors, but gql-tada doesn't seem to recognise this as valid usage.

Here's a minimal example demonstrating the pattern that's causing issues:

```typescript
const UserFragment = graphql(`
  fragment User on User {
    id
    name
    email
    createdAt
  }
`)

const { accessor } = createColumnHelper<ResultOf<typeof UserFragment>>()

// In React component
const columns = useMemo(() => [
  accessor((row) => row.name, {
    id: 'name',
    header: 'Name',
    cell: (info) => info.getValue()
  }),
  accessor((row) => row.email, {
    id: 'email',
    header: 'Email',
    cell: (info) => info.getValue()
  }),
  accessor((row) => row.createdAt, {
    id: 'createdAt',
    header: 'Created At',
    cell: (info) => formatDate(info.getValue())
  })
], [])
```

This pattern triggers warnings like "Field name is not used", "Field email is not used", etc., even though these fields are being accessed through the table column accessors.

Given the dynamic nature of these accessors, I suspect this might be fundamentally difficult for static analysis to track. I'm wondering if others have encountered this and if there are any recommended approaches. While disabling `trackFieldUsage` globally would solve this, it feels unfortunate to lose this valuable feature project-wide just for this specific pattern. Especially because I think the `trackFieldUsage` rule is an excellent feature that helps prevent over-fetching and maintains clean GraphQL queries.

**Comment by kitten** [comment]
There's only so many patterns that the `fieldUsage` can detect and it's certainly not exhaustive. If you have a local fragment that introduces various indirections to the type, then the `fieldUsage` tracker will lose track of the original type, and will be unable to detect that certain fields are used.

In complex cases like these, I'd recommend to simply turn off the `trackFieldUsage` feature, or to ignore its warnings. There's probably more work we can do to improve it, but that's pending some refactors
